disnake/poll.py

```diff
@@ -0,0 +1,415 @@
+        text = data.get("text")
+
+        emoji = None
+        if emoji_data := data.get("emoji"):
+            emoji = state._get_emoji_from_data(emoji_data)
+
+        return cls(text=text, emoji=emoji)
+
+    def _to_dict(self) -> PollCreateMediaPayload:
+        payload: PollCreateMediaPayload = {}
+        if self.text:
+            payload["text"] = self.text
+        if self.emoji:
+            if isinstance(self.emoji, (Emoji, PartialEmoji)):
+                if self.emoji.id:
+                    payload["emoji"] = {"id": self.emoji.id}
+                else:
+                    payload["emoji"] = {"name": self.emoji.name}
+            else:
+                payload["emoji"] = {"name": self.emoji}
+        return payload
+
+
+class PollAnswer:
+    """Represents a poll answer from discord.
+
+    .. versionadded:: 2.10
+
+    Parameters
+    ----------
+    media: :class:`PollMedia`
+        The media object to set the text and/or emoji for this answer.
+
+    Attributes
+    ----------
+    id: Optional[:class:`int`]
+        The ID of the answer that this object represents. This will be None only when the object
+        is builded manually. The library always provide this attribute.
+    media: :class:`PollMedia`
+        The media fields linked to this answer.
+    poll: Optional[:class:`Poll`]
+        The poll that contain this answer. This will be None only when the object
+        is builded manually. The library always provide this attribute.
+    count: :class:`int`
+        The number of votes for this answer.
+    self_voted: :class:`bool`
+        Whether the current user voted for this answer.
+    """
+
+    __slots__ = ("_state", "id", "media", "poll", "count", "self_voted")
+
+    def __init__(self, media: PollMedia) -> None:
+        self._state: Optional[ConnectionState] = None
+        self.id: Optional[int] = None
+        self.poll: Optional[Poll] = None
+        self.media = media
+        self.count: int = 0
+        self.self_voted: bool = False
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__} poll_media={self.media!r}>"
+
+    @classmethod
+    def from_dict(cls, state: ConnectionState, data: PollAnswerPayload) -> PollAnswer:
+        answer = cls(PollMedia.from_dict(state, data["poll_media"]))
+        answer.id = int(data["answer_id"])
+        answer._state = state
+
+        return answer
+
+    def _to_dict(self) -> PollCreateAnswerPayload:
+        return {"poll_media": self.media._to_dict()}
+
+    def voters(
+        self, *, limit: Optional[int] = 100, after: Optional[Snowflake] = None
+    ) -> PollAnswerIterator:
+        """Returns an :class:`AsyncIterator` representing the users that have voted for this answer.
+
+        The ``after`` parameter must represent a member and meet the :class:`abc.Snowflake` abc.
+
+        .. note::
+
+            This method works only on PollAnswer(s) objects that are fetched from the API and not on the ones built manually.
```

Comment:
```suggestion
            This method works only on PollAnswer(s) objects that originate from the API and not on the ones built manually.
```
just to make this clear, since receiving a poll via gateway isn't technically "fetching"